### ISSUE TITLE
Use field order by default?

### ISSUE THREAD

**Issue by kesne** [kesne/graphql-rpc#1]
Right now, we require every field to have a `@field` directive that provides a number to tag the field. Because schema fields are actually ordered, we could use the definition order in the schema to automatically number the fields.

If you want to remove a field, we could use `@deprecated`, or fall back into manually defining the field numbers via `@field`.

To reserve fields for later use, we could fall back into manually defining field numbers, and using the `@reserved` object directive.

**Comment by Emily** [comment]
This would mean changing field order in the schema would be a breaking change to the serialization until said client refetches the schema.

**Comment by kesne** [comment]
@Emily That's correct. FWIW you have the same issue if you just assign different field numbers in the schema too. I guess that's probably less likely.

I imagine no matter what we do, we'll create some lint tooling to catch backwards-incompatible changes.
